### PR TITLE
enh(UI): Hide empty graphs from service details

### PR DESCRIPTION
**Pull request by UrBnW**
Hi,

## Description

One may intentionally disable graphs for one or more services, or for its whole installation, switching of RRDs processing.
This then leads to an empty "_There's no data_" graphs in the _Services Status Details_ pages, which is useless and can (well does) confuse users.
Let's then get rid of these empty graphs when there's no data (from `data_bin`) to plot.

Fixes : N/A (no issue has been opened for this, I directly submitted the corresponding PR)

## Type of change

Well, hard to tell, between the 2 followings :
- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Disable RRD processing (be sure then your `data_bin` table is empty), or at least remove graphs / empty `data_bin` table for a service, and verify, in the _Services Status Details_ pages, that the following empty graph does not show up :

image

Thank you 👍

Edit : issue still present in 19.10.10.

### CODE REVIEW

**Comment by adr-mo** [comment]
Hi @UrBnW ,

Thanks for the contribution :100:

Your contribution will be discussed during the next weekly sessions of refinement with the team.

Thanks :+1:

Regards,

**Comment by kduret** [comment]
Note : need to check if it can introduce performance issue related to big data_bin table

**Comment by UrBnW** [comment]
> Note : need to check if it can introduce performance issue related to big data_bin table

Finally the relation is on the `metrics` table, not the `data_bin` table, I updated the PR accordingly.
There should then not be any performance issue 👍

File: www/include/monitoring/objectDetails/serviceDetails.php [hunk 1]
```diff
@@ -533,8 +533,8 @@
 
         $optionsURL = "host_name=" . urlencode($host_name) . "&service_description=" . urlencode($svc_description);
 
-        $query = "SELECT id FROM `index_data` WHERE host_name = '" . $pearDBO->escape($host_name) .
-            "' AND service_description = '" . $pearDBO->escape($svc_description) . "' LIMIT 1";
+        $query = "SELECT id FROM `index_data`,`metrics` WHERE host_name = '" . $pearDBO->escape($host_name) .
```
Comment (cgagnaire) [inline www/include/monitoring/objectDetails/serviceDetails.php]:
```suggestion
        $query = "SELECT id FROM `index_data`, `metrics` WHERE host_name = '" . $pearDBO->escape($host_name) .
```

File: www/include/monitoring/objectDetails/serviceDetails.php [hunk 2]
```diff
@@ -533,8 +533,8 @@
...
+            "' AND service_description = '" . $pearDBO->escape($svc_description) . "' AND id=index_id LIMIT 1";
```
Comment (cgagnaire) [inline www/include/monitoring/objectDetails/serviceDetails.php]:
```suggestion
            "' AND service_description = '" . $pearDBO->escape($svc_description) . "' AND id = index_id LIMIT 1";
```

File: www/include/monitoring/objectDetails/serviceDetails.php (hunk 1 above)
Comment (UrBnW) [reply www/include/monitoring/objectDetails/serviceDetails.php]: Thx for your review @cgagnaire, just corrected 👍

File: www/include/monitoring/objectDetails/serviceDetails.php (hunk 2 above)
Comment (UrBnW) [reply www/include/monitoring/objectDetails/serviceDetails.php]: Done.

**Comment by UrBnW** [comment]
There are 2 following tiny PRs which are in some kinds related to this :
https://github.com/centreon/centreon-widget-host-monitoring/pull/73
https://github.com/centreon/centreon-widget-service-monitoring/pull/130
If we can merge them at the same time, would be perfect 👍
Thank you !

**Comment by sc979** [comment]
Hi @UrBnW  ,

Your work has been selected by the Team and moved to a new internal branch, so we can integrate it to our workflow.

Stay posted here for status updates.

Regards.
sc979

**Comment by sc979** [comment]
MON-4928

**Comment by UrBnW** [comment]
Did you internally managed to look at these tiny PRs ?
Many thanks 👍

**Comment by sc979** [comment]
Hi @UrBnW ,
And, sorry for the delay, but we're overwhelmed right now.
Your modifications are in the backlog under the reference MON-4928
Best regards,
sc979

**Comment by UrBnW** [comment]
Good news, many thanks @cgagnaire 👍

**Pull request merged**